File: plugins/capawesome-capacitor-bluetooth-low-energy/ios/Plugin/Classes/Options/ReadDescriptorOptions.swift
import Foundation
import Capacitor

@objc public class ReadDescriptorOptions: NSObject {
    private var characteristicId: String
    private var descriptorId: String
    private var deviceId: String
    private var serviceId: String

    init(characteristicId: String, descriptorId: String, deviceId: String, serviceId: String) {
        self.characteristicId = characteristicId
        self.descriptorId = descriptorId
        self.deviceId = deviceId
        self.serviceId = serviceId
    }

    func getCharacteristicId() -> String {
        return characteristicId
    }

    func getDescriptorId() -> String {
        return descriptorId
    }

    func getDeviceId() -> String {
        return deviceId
    }

    func getServiceId() -> String {
        return serviceId
    }
}

File: plugins/capawesome-capacitor-bluetooth-low-energy/android/src/main/java/io/capawesome/capacitorjs/plugins/bluetoothle/classes/options/ReadDescriptorOptions.java
package io.capawesome.capacitorjs.plugins.bluetoothle.classes.options;

import androidx.annotation.NonNull;

public class ReadDescriptorOptions {

    @NonNull
    private String characteristicId;

    @NonNull
    private String descriptorId;

    @NonNull
    private String deviceId;

    @NonNull
    private String serviceId;

    public ReadDescriptorOptions(
        @NonNull String characteristicId,
        @NonNull String descriptorId,
        @NonNull String deviceId,
        @NonNull String serviceId
    ) {
        this.characteristicId = characteristicId;
        this.descriptorId = descriptorId;
        this.deviceId = deviceId;
        this.serviceId = serviceId;
    }

    @NonNull
    public String getCharacteristicId() {
        return characteristicId;
    }

    @NonNull
    public String getDescriptorId() {
        return descriptorId;
    }

    @NonNull
    public String getDeviceId() {
        return deviceId;
    }

    @NonNull
    public String getServiceId() {
        return serviceId;
    }
}
